packages/core/src/formats.ts

```diff
@@ -75,9 +75,9 @@ function getMemoized<T>(getKey: () => string, construct: () => T) {
 
 function cacheKey(
   type: string,
-  locales?: readonly string[],
-  options: unknown = {}
+  locales: readonly string[],
+  options?: Intl.DateTimeFormatOptions | Intl.NumberFormatOptions
```

Comment:
maybe default to `{}` as before, the differences are:

```
JSON.stringify(undefined)
undefined
JSON.stringify({})
'{}'
```

so it affects the generated cache key, unless you intend to make `cacheKey(type, locales)` to generate a different key if used `cacheKey(type, locales, {})` ?

Reply:
yes, I can revert that part

> unless you intend to make cacheKey(type, locales) to generate a different key compared to cacheKey(type, locales, {})

no, that is not an intent but a side effect. I thought, "how are people going to call this"? And I think it's unlikely they would call `i18n. number(123, {})`. However, if they do call `i18n. number(123, {})` and `i18n. number(123})` then it's true that the cache will have 2 entries for the same number formatter which is not great _but_ I also thought about the default parameter `options = {}` which means that every time `i18n. number(123})` is called there is an empty object allocated and disposed. It's not important though, I can revert this part...